elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/dynamic/DynamicAccessorCompanion.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.converters.avro.dynamic
+
+import com.spotify.elitzur.MetricsReporter
+import com.spotify.elitzur.converters.avro.dynamic.DynamicAccessorException.InvalidModifierException
+import com.spotify.elitzur.converters.avro.AvroElitzurConversionUtils.byteBufferToByteArray
+import com.spotify.elitzur.validators.{
+  BaseCompanion,
+  BaseValidationType,
+  SimpleCompanionImplicit,
+  Validator
+}
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+import scala.reflect.runtime.universe.typeOf
+
+import java.{util => ju}
+import collection.JavaConverters._
+
+//scalastyle:off line.size.limit structural.type
+class DynamicAccessorCompanion[T: TypeTag, LT <: BaseValidationType[T]: ClassTag: ({type L[x] = SimpleCompanionImplicit[T, x]})#L](
+  implicit metricReporter: MetricsReporter) {
+//scalastyle:on line.size.limit structural.type
+
+  private val companion: BaseCompanion[T, LT] =
+    implicitly[SimpleCompanionImplicit[T, LT]].companion
+  private[dynamic] val validationType: String = companion.validationType
+
+  private[dynamic] def getModifier(input: String): Modifier = {
+    if (!input.contains("[")) {
+      NoModifier
+    } else {
+      val validatorModifierStrArray: Array[String] = input.split("\\[").dropRight(1)
+
+      validatorModifierStrArray.mkString(".") match {
+        case "Option" => OptModifier
+        case "Seq" => SeqModifier
```

Comment:
I arbitrarily picked `Seq` to be the keyword to represent an array. This can be anything. Should we change it to `Array` or `Repeated`?